runtime-light/stdlib/array/array-functions.h

```diff
@@ -221,19 +424,108 @@ array<T> f$array_combine(const array<T1> &keys, const array<T> &values) {
+    case SORT_STRING:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::rsort_compare_string<T>(), true);
+    default:
+      php_warning("Unsupported sort_flag in function rsort");
+  }
 }
 
-template<class T, class T1>
-void f$uksort(array<T> &a, const T1 &compare) {
-  php_critical_error("call to unsupported function");
+template<class T>
+task_t<void> f$arsort(array<T> &a, int64_t flag = SORT_REGULAR) {
+  switch (flag) {
+    case SORT_REGULAR:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::rsort_compare<T>(), false);
+    case SORT_NUMERIC:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::rsort_compare_numeric<T>(), false);
+    case SORT_STRING:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::rsort_compare_string<T>(), false);
+    default:
+      php_warning("Unsupported sort_flag in function arsort");
+  }
+}
+
+template<class T>
+task_t<void> f$ksort(array<T> &a, int64_t flag = SORT_REGULAR) {
+  switch (flag) {
+    case SORT_REGULAR:
+      co_return co_await array_functions_impl_::ksort<task_t<void>>(a, array_functions_impl_::sort_compare<typename array<T>::key_type>());
+    case SORT_NUMERIC:
+      co_return co_await array_functions_impl_::ksort<task_t<void>>(a, array_functions_impl_::sort_compare_numeric<typename array<T>::key_type>());
+    case SORT_STRING:
+      co_return co_await array_functions_impl_::ksort<task_t<void>>(a, array_functions_impl_::sort_compare_string<typename array<T>::key_type>());
+    default:
+      php_warning("Unsupported sort_flag in function ksort");
+  }
+}
+
+template<class T>
+task_t<void> f$krsort(array<T> &a, int64_t flag = SORT_REGULAR) {
+  switch (flag) {
+    case SORT_REGULAR:
+      co_return co_await array_functions_impl_::ksort<task_t<void>>(a, array_functions_impl_::rsort_compare<typename array<T>::key_type>());
+    case SORT_NUMERIC:
+      co_return co_await array_functions_impl_::ksort<task_t<void>>(a, array_functions_impl_::rsort_compare_numeric<typename array<T>::key_type>());
+    case SORT_STRING:
+      co_return co_await array_functions_impl_::ksort<task_t<void>>(a, array_functions_impl_::rsort_compare_string<typename array<T>::key_type>());
+    default:
+      php_warning("Unsupported sort_flag in function krsort");
+  }
+}
+
+template<class T>
+task_t<void> f$asort(array<T> &a, int64_t flag = SORT_REGULAR) {
+  switch (flag) {
+    case SORT_REGULAR:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::sort_compare<T>(), false);
+    case SORT_NUMERIC:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::sort_compare_numeric<T>(), false);
+    case SORT_STRING:
+      co_return co_await array_functions_impl_::sort<task_t<void>>(a, array_functions_impl_::sort_compare_string<T>(), false);
```

Comment:
Why is it interruptible?
I thought that all sort function that are not receiving callable as argument should **not** be interruptible.
Am I wrong?

Reply:
You are right
I did it interruptible for easy implementation. But maybe it would be better to make it a right way in this pr

Reply:
If it's a big deal then it's OK to leave as is.

Reply:
fix